components/credentials/src/main/java/org/cloudfoundry/credhub/generators/CertificateGenerator.java

```diff
@@ -23,17 +24,23 @@
   private final RsaKeyPairGenerator keyGenerator;
   private final SignedCertificateGenerator signedCertificateGenerator;
   private final CertificateAuthorityService certificateAuthorityService;
+  private final Integer caMinimumDuration;
+  private final Integer leafMinimumDuration;
```

Comment:
I would like to see these variables to be of type [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html). 

In the configuration, we don't have an option and have to use Strings and Integer, but once the middleware is passed, I think that it's better to use types that represent better what the object is. 
It also solves the problem of the variable name not expressing which unit the integer is in.

Reply:
While I'm generally in favor of better types, we don't think it adds a great deal of value in this case. The two "duration" fields are only used to overwrite the pre-existing `duration` field on the `CertificateGenerationParameters` class. That field comes in via the API and is already an integer in units of days:
https://github.com/cloudfoundry-incubator/credhub/blob/3a71bf0a2f00e8d031eb48569ddd083077ccbaf5/components/credentials/src/main/kotlin/org/cloudfoundry/credhub/domain/CertificateGenerationParameters.kt#L41

If we were to convert it to a Duration, we would just be immediately converting it back to an integer when we do set it on the generation parameters. See:
https://github.com/cloudfoundry-incubator/credhub/blob/65c2c17153bee5bb8ccfae307c83734e3000f116/components/credentials/src/main/java/org/cloudfoundry/credhub/generators/CertificateGenerator.java#L133-L139
and
https://github.com/cloudfoundry-incubator/credhub/blob/65c2c17153bee5bb8ccfae307c83734e3000f116/components/credentials/src/main/java/org/cloudfoundry/credhub/generators/CertificateGenerator.java#L61

The variables are not used anywhere else that would benefit from the more rich datatype. We've renamed the variables to include `InDays` to make units more clear. Let us know if that isn't sufficient and we can update it to be an actual `Duration` object.

Reply:
Hmm alright, indeed. I guess I was hoping that there was some intermediate layer in which we could have put the Duration. It looks kinda superfluous here. Although I think that I would have still done it myself, because I like high level types. Your call :).